daemon/tests/happy_path.rs

```diff
@@ -101,39 +84,13 @@ async fn taker_takes_order_and_maker_accepts_and_contract_setup() {
     taker.take_order(received.clone(), Usd::new(dec!(5)));
     let (_, _) = next_cfd(&mut taker.cfd_feed, &mut maker.cfd_feed).await;
 
-    maker
-        .mocks
-        .oracle()
-        .await
-        .expect_get_announcement()
-        .returning(|_| Some(dummy_announcement()));
-
-    taker
-        .mocks
-        .oracle()
-        .await
-        .expect_get_announcement()
-        .returning(|_| Some(dummy_announcement()));
+    mock_oracle_annoucement(&mut maker.mocks.oracle().await);
```

Comment:
What about:

```suggestion
    maker.mocks.mock_oracle_announcement();
```

Reply:
cool! did it for all of them. accessing mutexguard is `async`, so I had to `.await` all of these calls - but they still look nicer than before :) and it should be easier to discover the calls with the IDE.